src/github.com/getlantern/fronted/direct.go

```diff
@@ -136,6 +137,13 @@ func (d *direct) Do(req *http.Request) (*http.Response, error) {
 // Dial persistently dials masquerades until one succeeds.
 func (d *direct) Dial(network, addr string) (net.Conn, error) {
 	gotFirst := false
+	candidateCh := <-candidateChCh
+	select {
+	case candidateChCh <- candidateCh:
```

Comment:
@myleshorton 

The following sequence will break this:

1. Configure() pushes candidateCh A
2. Dial() pops candidateCh A
3. Configure() pushes candidateCh B (this will succeed because there's room in the channel)
4. Another Dial() pops candidateCh B
5. The first Dial() pushes candidateCh A
6. The second Dial() attempts to push candidateCh B, but the channel is full so it discards it

Now, subsequence dials using candidateCh A even though we want them to use candidateCh B.

You mentioned that Dial() can happen before Configure(), which keeps atomic.Value from working.  You can use eventual.Value for this.